### PR TITLE
Order total authorized performance

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it increases performance for order totalAuthorized field.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2894 Optimize totalAuthorized resolver in Order type](https://app.clickup.com/t/2549495/SALEOR-2894)

File: saleor/order/models.py [hunk 1]
```diff
@@ -342,10 +343,7 @@ def can_mark_as_paid(self):
 
     @property
     def total_authorized(self):
-        payment = self.get_last_payment()
```
Comment (maarcingebala) [inline saleor/order/models.py:345]: We should also look at other usages of `get_last_payment` from Order model and replace it with loading payments with dataloader and using the utility function that this PR adds. I'm not saying we have to do it in this PR but we should keep it on the radar.

Comment (maarcingebala) [reply saleor/order/models.py:345]: Btw is the `total_authorized` property needed anymore after migrating to dataloader in the Order type?

**Review by IKarbowiak — APPROVED** [review]

File: saleor/payment/utils.py [hunk 2]
```diff
@@ -365,3 +367,14 @@ def is_currency_supported(currency: str, gateway_id: str, manager: "PluginsManag
     """Return true if the given gateway supports given currency."""
     available_gateways = manager.list_payment_gateways(currency=currency)
     return any([gateway.id == gateway_id for gateway in available_gateways])
+
+
+def get_last_payment(payments: List[Payment]):
```
Comment (korycins) [inline saleor/payment/utils.py:372]: It looks like it returns `last_payment` for the order but we never check if the payment is valid.  Not sure if this a proper way

**Review by fowczarek — APPROVED** [review]

File: saleor/order/models.py (hunk 1 above)
Comment (tomaszszymanski129) [reply saleor/order/models.py:345]: @maarcingebala About property - It is used but on subsequent PRs i'll probably be able to get rid of it, i also have it in mind, it's just used in some other properties which will also be changed to use DLs.

**Review by maarcingebala — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Pull request merged**